pkg/acsengine/defaults.go

```diff
@@ -209,6 +220,24 @@ func setDefaultCerts(a *api.Properties) (bool, error) {
 		a.CertificateProfile.SetCAPrivateKey(caPair.PrivateKeyPem)
 	}
 
+	// Work out the first IP in KubeServiceCidr range
+	_, serviceCidr, err := net.ParseCIDR(a.OrchestratorProfile.KubernetesConfig.ServiceCIDR)
+	if err != nil {
+		return false, err
+	}
+	// Based off https://play.golang.org/p/m8TNTtygK0
+	// Increment the last octet if it's zero.
+	firstIP := func(ip net.IP) net.IP {
+		for j := len(ip) - 1; j >= 0; j-- {
+			if ip[j] == 0 {
+				ip[j]++
```

Comment:
I can now having played with it a bit more.

There is a service called `kubernetes` created in every cluster, and it has as it's IP the first IP in the service CIDR range. By default this is 10.0.0.1 (the default range is 10.0.0.0/16). Previously this was handled in CreatePki (and removed in this diff in the only changes in `pkg/acsengine/pki.go`), but rather than passing another argument in for it to then add to the IPs I handle it here.

The reason I do the ParseCIDR and then add the net is to be more lenient in the input - for instance this would cope with:

- 192.168.1.128/25 (192.168.1.128 would be the serviceIp)
- 192.168.1.1/25 or 192.168.1.0/25, both yielding 192.168.1.1 as the service IP etc.